bluepyentity/entity_definitions.py

```diff
@@ -0,0 +1,385 @@
+    wasInformedBy: IDConverter = None
+    wasInfluencedBy: IDConverter = None
+
+
+class AnalysisReport(Entity):
+    image: ListOfPath = None
+
+    # Added
+    derivation: IDConverter = None
+    types: ListOfStr = None
+
+    def _get_elements_to_attach(self, definition):
+        to_attach = super()._get_elements_to_attach(definition)
+        to_attach.update({k: definition.pop(k, None) for k in ("image", "derivation")})
+
+        return to_attach
+
+    def _attach_to_resource(self, forge, resource, to_attach):
+        super()._attach_to_resource(forge, resource, to_attach)
+
+        if to_attach["image"] is not None:
+            for item in to_attach["image"]:
+                resource.add_files(item)
+
+        if to_attach["derivation"] is not None:
+            derivation = _fetch(to_attach["derivation"], forge)
+            resource.add_derivation(derivation)
+
+
+class BrainRegion(BaseModel):
+    id: pydantic.StrictStr
+
+    def get_formatted_definition(self, forge):
+        return forge.reshape(_fetch(self.id, forge), ["id", "label", "notation"])
+
+
+class BrainRegionConverter(Converter):
+    @classmethod
+    def _custom_validator(cls, item):
+        if isinstance(item, str):
+            item = {"id": item}
+        if isinstance(item, dict):
+            return BrainRegion.from_dict(item)
+
+        raise TypeError("str or dict type expected")
+
+
+class BrainLocation(BaseModel):
+    brainRegion: BrainRegionConverter
+
+
+class ModelInstance(Entity):
+    modelOf: str = None
+    brainLocation: BrainLocation = None
+    subject: IDConverter = None
+
+
+class DetailedCircuit(ModelInstance):
+    circuitConfigPath: DataDownload
+    circuitType: str = None
+
+    # Added
+    atlasRelease: IDConverter = None
+
+
+class DetailedCircuitValidation(Activity):
+    pass
+
+
+class DetailedCircuitValidationReport(AnalysisReport):
+    pass
+
+
+class Simulation(Activity):
+    spikes: IDConverter = None
+    jobId: str = None
+    path: pathlib.Path = None
+    params: str = None
+
+    # Added
+    simulationConfigPath: DataDownload
+
+
+class SimulationCampaignGeneration(Activity):
+    pass
+
+
+class SimulationConfiguration(Entity):
+    circuit: IDConverter = None
+
+
+class SimulationCampaignConfiguration(Entity):
+    configuration: pathlib.Path = None
+    template: pathlib.Path = None
+    target: pathlib.Path = None
+
+    def _get_elements_to_attach(self, definition):
+        to_attach = super()._get_elements_to_attach(definition)
+        to_attach.update(
+            {k: definition.pop(k, None) for k in ("configuration", "template", "target")}
+        )
+
+        return to_attach
+
+    def _attach_to_resource(self, forge, resource, to_attach):
+        super()._attach_to_resource(forge, resource, to_attach)
+
+        for file_key in ("configuration", "template", "target"):
+            if to_attach[file_key] is not None:
+                setattr(resource, file_key, forge.attach(path=to_attach[file_key]))
```

Comment:
`content_type` needs be defined in forge.attach otherwise everything will be registered as octet-stream.

Reply:
I wonder what should be the content type of these files. 
Also, note that at least some of the known formats are automatically recognized:
* `json` as `application/json`
* `png` as `image/png`

Reply:
I think what I'd do with these three files (as they're text files) is that I'd fallback to `text/plain` if it can't be determined from the file type. I just hope that the guesstimation of the content-type is done in nexus-forge, not in the backend 🤞

Reply:
nexus-forge estimates the mime-type from filepath with mimetypes. Added similar functionality to bluepyentity with a possibility have a fallback if it can't be determined.

Reply:
Nice!